pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
 	if err != nil {
 		return err
 	}
-	return c.Do(request, 200, nil)
+	return c.Do(request, "200", nil)
+}
+
+//MCServerRolesInfo is the struct for the server roles of the localhost, in this case SplunkMonitoringConsole
+type MCServerRolesInfo struct {
+	ServerRoles []string `json:"server_roles"`
+}
+
+//MCDistributedPeers is the struct for information about distributed peers of the monitoring console
+type MCDistributedPeers struct {
+	ClusterLabel []string `json:"cluster_label"`
+	ServerRoles  []string `json:"server_roles"`
+}
+
+//AutomateMCApplyChanges change the state of new indexers from "New" to "Configured" and add them in monitoring console asset table
+func (c *SplunkClient) AutomateMCApplyChanges(mock bool) error {
+	if mock {
+		return nil
+	}
+	var configuredPeers, indexerMemberList, licenseMasterMemberList string
+	apiResponseServerRoles, err := c.GetMonitoringconsoleServerRoles()
+	if err != nil {
+		return err
+	}
+
+	apiResponseMCDistributedPeers := struct {
+		Entry []struct {
+			Name    string             `json:"name"`
+			Content MCDistributedPeers `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/services/search/distributed/peers"
+	err = c.Get(path, &apiResponseMCDistributedPeers)
+	if err != nil {
+		return err
```

Comment:
My ask was different to have a meaningful error based on the situation than a generic error. Anyhow, after going through a couple of error messages, it seems that SDK is also printing the line numbers as part of the stack trace, which has the line numbers, so we are good here.

Reply:
https://github.com/splunk/splunk-operator/pull/135/files/c04d58a94806ee435c5fda45ea95a264aeff76b0#diff-f37c2254cc2ca855cf4bea2777b7e7c0R79 This will give the request URL and response code received, which I think should be sufficient to debug